### PR TITLE
[generate-key] Simplify generate-key and generalize around it

### PR DESCRIPTION
**Pull request by davidiw**
We benefit by having a single crate for generating, storing, and loading keys in LCS This will help prevent accidental breakage in configuration / code. So pushed generate-keypair into that direction. This may not be a full featured tool, but it solves all of our needs for now. I'll probably extend it in the future as needed.

### CODE REVIEW

File: client/cli/src/client_proxy.rs [hunk 1]
```diff
@@ -132,15 +132,12 @@ impl ClientProxy {
         let faucet_account = if faucet_account_file.is_empty() {
             None
         } else {
-            let faucet_account_keypair: KeyPair<Ed25519PrivateKey, Ed25519PublicKey> =
```
Comment (mimoo) [inline client/cli/src/client_proxy.rs:135]: I think we should keep the keypair and go this way: https://github.com/libra/libra/pull/3414

(only serialize/deserialize the private key pair, generate the public part at loading/creation)

File: config/generate-key/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+use std::{
+    fs::{self, File},
+    io::Write,
+    path::Path,
+};
+
+pub fn generate_key() -> Ed25519PrivateKey {
+    let mut seed_rng = OsRng;
+    let mut rng = rand::rngs::StdRng::from_seed(seed_rng.gen());
+    Ed25519PrivateKey::generate(&mut rng)
```
Comment (mimoo) [inline config/generate-key/src/lib.rs:17]: never seed an RNG yourself: use [ThreadRng](https://docs.rs/rand/0.7.3/rand/rngs/struct.ThreadRng.html) instead

File: client/cli/src/client_proxy.rs (hunk 1 above)
Comment (davidiw) [reply client/cli/src/client_proxy.rs:135]: I would agree if we ever used the public key, but we don't.

File: config/generate-key/src/lib.rs (hunk 2 above)
Comment (davidiw) [reply config/generate-key/src/lib.rs:17]: This is just a code mod generate-keypair -> generate-key

Your argument makes sense though... but out of scope for this PR.

**Comment by mimoo** [comment]
please check my comments : o 

@bors-libra delegate+

**Pull request closed without merging**